fondant/cli.py

```diff
@@ -170,40 +68,198 @@ def pipeline_from_string(import_string: str) -> Pipeline:
+            parser.print_help()
+            sys.exit(0)
+        return parser.parse_known_args(input_args)
+
+    def explore(self, input_args: t.List[str]):
+        parser = argparse.ArgumentParser()
+        parser.add_argument(
+            "--data-directory",
+            "-d",
+            help="""Path to the source directory that contains the data produced
+            by a fondant pipeline.""",
+            required=False,
+            type=str,
+        )
+        parser.add_argument(
+            "--container",
+            "-r",
+            default=DEFAULT_CONTAINER,
+            help="Docker container to use. Defaults to ghcr.io/ml6team/data_explorer.",
+        )
+        parser.add_argument(
+            "--tag",
+            "-t",
+            default=DEFAULT_TAG,
+            help="Docker image tag to use.",
+        )
+        parser.add_argument(
+            "--port",
+            "-p",
+            default=DEFAULT_PORT,
+            type=int,
+            help="Port to expose the container on.",
+        )
+        parser.add_argument(
+            "--credentials",
+            "-c",
+            help="""Path mapping of the source (local) and target (docker file system)
+                credential paths in the format of src:target
+                \nExamples:\n
+                Google Cloud: $HOME/.config/gcloud/application_default_credentials.json:/root/."
+                + "config/gcloud/application_default_credentials.json
+                AWS: HOME/.aws/credentials:/root/.aws/credentials
+                More info on
+                Google Cloud:
+                https://cloud.google.com/docs/authentication/application-default-credentials
+                AWS: https: // docs.aws.amazon.com/sdkref/latest/guide/file-location.html
+            """,
+        )
+        args = parser.parse_args(input_args)
+
+        if not args.data_directory:
+            logging.warning(
+                "You have not provided a data directory."
+                + "To access local files, provide a local data directory"
+                + " with the --data-directory flag.",
+            )
+        else:
+            logging.info(f"Using data directory: {args.data_directory}")
+            logging.info(
+                "This directory will be mounted to /artifacts in the container.",
+            )
+
+        if not args.credentials:
+            logging.warning(
+                "You have not provided a credentials file. If you wish to access data "
+                "from a cloud provider, mount the credentials file with the --credentials flag.",
+            )
+
+        if not shutil.which("docker"):
+            logging.error(
+                "Docker runtime not found. Please install Docker and try again.",
+            )
+
+        run_explorer_app(**vars(args))
+
+    def compile(self, input_args):
+        parser = argparse.ArgumentParser(add_help=False)
+
+        parser.add_argument(
+            "mode",
```

Comment:
I think it's a more logical user interface if this is a flag instead of a subcommand.

So:
```
fondant compile <pipeline> --local
```
instead of
```
fondant compile local <pipeline>
```

I think this can be achieved in the current implementation. We might just have to lift the `pipeline` parameter from `compile_{mode}` to this function, which I think makes sense.